box2dworld.cpp

```diff
@@ -137,6 +139,8 @@ Box2DWorld::Box2DWorld(QObject *parent) :
 
 {
     mWorld.SetDestructionListener(this);
+    if(!defaultWorld)
+        defaultWorld = this;
 }
 
 Box2DWorld::~Box2DWorld()
```

Comment:
Probably the destructor should check if `this == defaultWorld` and unset it if so.